.github/workflows/CI.yml

```
@@ -37,11 +39,16 @@ jobs:
           path: ~/.julia/artifacts
           key: ${{ runner.os }}-test-artifacts-${{ hashFiles('**/Project.toml') }}
           restore-keys: ${{ runner.os }}-test-artifacts
-      - uses: julia-actions/julia-buildpkg@v1
+      - name: "Install Makie and instantiate project"
+        shell: julia --color=yes --project {0}
+        run: |
+          using Pkg
+          Pkg.add(Pkg.PackageSpec(; name="Makie", version="${{ matrix.makie }}"))
+          Pkg.instantiate()
```

just FYI, we use this trick a lot, but I recently learned `Pkg.test` can resolve you into a different version if there's compatibility problems, which is kind of unfortunate. I _think_ `Pkg.develop` might work, since I'm not sure if Pkg can switch you out then

---

the other thing you can do is dynamically rewrite compat to force the version you want... I think a helper GHA could be nice to do that, where you put a package and a version and it forces it

---

Ah, yeah, I think `develop` forces things.


```suggestion
        run: |
          using Pkg
          Pkg.develop(Pkg.PackageSpec(; name="Makie", version="${{ matrix.makie }}"))
          Pkg.instantiate()
```